File: submitted_code/237.delete-node-in-a-linked-list.cpp
//
// @lc app=leetcode.cn id=237 lang=cpp
//
// [237] delete-node-in-a-linked-list
//
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    void deleteNode(ListNode* node) {
        ListNode *he = node->next;
        node->val = he->val;
        node->next = he->next;
        delete(he);
    }
};
// @lc code=end

File: submitted code/237.delete-node-in-a-linked-list.java
//
// @lc app=leetcode.cn id=237 lang=java
//
// [237] delete-node-in-a-linked-list
//
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
        ListNode p = node.next;
        while(p.next != null){
            node.val = p.val;
            node = node.next;
            p = p.next;
        }
        node.val = p.val;
        node.next = null;
    }
}
// @lc code=end